internal/constraints/constraints.go

```diff
@@ -315,19 +385,35 @@ func IsConstrained(constraint projectfile.Constraint) (bool, int) {
 // FilterUnconstrained filters a list of constrained entities and returns only
 // those which are unconstrained. If two items with the same name exist, only
 // the most specific item will be added to the results.
-func FilterUnconstrained(items []projectfile.ConstrainedEntity) []projectfile.ConstrainedEntity {
+func FilterUnconstrained(conditional *Conditional, items []projectfile.ConstrainedEntity) ([]projectfile.ConstrainedEntity, error) {
 	type itemIndex struct {
 		specificity int
 		index       int
 	}
 	selected := make(map[string]itemIndex)
 
+	if conditional == nil {
+		logging.Error("FilterUnconstrained called with nil conditional")
+	}
+
 	for i, item := range items {
-		c := item.ConstraintsFilter()
-		constrained, specificity := IsConstrained(c)
-		if !constrained {
-			if s, exists := selected[item.ID()]; !exists || s.specificity < specificity {
-				selected[item.ID()] = itemIndex{specificity, i}
+		if conditional != nil && item.ConditionalFilter() != "" {
```

Comment:
Maybe we could print an error message, if both `ConditionalFilter` and `ConstraintsFilter` are provided.  I think it is the correct way to not mix them together, as we are deprecating the "constraints" anyways.  But if a user has defined both, they might wonder why their `constraints` are being ignored...

Reply:
This is easier said than done since `print` is deprecated and passing `outputer` in to this level would be pretty awkward. We could pre-validate the activestate.yaml for this type of stuff but that feels like overkill for the low severity of this use-case.